app/components/Cart.js

```diff
@@ -1,78 +1,79 @@
-} else{
-    displayList = cartList.map(orderable => {
-       // var keykey=orderable.dict+orderable.id;
-        if(0) { 
-            return(
-            <Orderable 
-                orderableDetails={orderable} 
-                key={orderable.id}
-                addToCart={this.props.addToCart}
-                removeFromCart={this.props.removeFromCart}
-                cartView={true}
-                checkDuplicate = {this.props.checkDuplicate}
-                checkWarning = {this.props.checkWarning}
-                checkListOrderable = {this.props.checkListOrderable}
-                checkListSame = {this.props.checkListSame}
+    if (cartList.length === 0) {
+      displayList = "No orders selected";
+    } else {
+      displayList = cartList.map(orderable => {
+        // var keykey=orderable.dict+orderable.id;
+        if (0) {
+          return (
+            <Orderable
+              orderableDetails={orderable}
+              key={orderable.id}
+              addToCart={this.props.addToCart}
+              removeFromCart={this.props.removeFromCart}
+              cartView
+              checkDuplicate = {this.props.checkDuplicate}
+              checkWarning = {this.props.checkWarning}
+              checkListOrderable = {this.props.checkListOrderable}
+              checkListSame = {this.props.checkListSame}
             />
-        )
-        }else {
-            var orderGroup = orderable
-            return(
-            <OrderableGroup 
-                groupDetails={orderGroup} 
-                key={orderGroup.id}
-                addToCart={this.props.addToCart}
-                removeFromCart={this.props.removeFromCart}
-                duplicateError={duplicateError}
-                cartView={true}
-                checkDuplicate = {this.props.checkDuplicate}
-                checkWarning = {this.props.checkWarning}
-                checkListOrderable = {this.props.checkListOrderable}
-                checkListSame = {this.props.checkListSame}
-                removeGroupFromCart={this.props.removeGroupFromCart}
+          );
+        } else {
+          const orderGroup = orderable;
+          return (
+            <OrderableGroup
+              groupDetails={orderGroup}
+              key={orderGroup.id}
+              addToCart={this.props.addToCart}
+              removeFromCart={this.props.removeFromCart}
+              duplicateError={duplicateError}
+              cartView
+              checkDuplicate = {this.props.checkDuplicate}
+              checkWarning = {this.props.checkWarning}
+              checkListOrderable = {this.props.checkListOrderable}
+              checkListSame = {this.props.checkListSame}
+              removeGroupFromCart={this.props.removeGroupFromCart}
             />
-            )
+          );
         }
-    })
-}
-      
-//might be nice to arrange the cartList by Ordertype?
+      });
+    }
+
+    //might be nice to arrange the cartList by Ordertype?
```

Comment:
There is something weird going on with the ordering right now where things get rearranged when you add or remove items. What's up with that?